src/main/java/org/assertj/core/api/AbstractDateAssert.java

```diff
@@ -2188,6 +2188,51 @@ public S hasSameTimeAs(Date date) {
   }
 
   /**
+   * Same assertion as {@link #hasSameTimeAs(Date)} but given date is represented as String either with one of the
+   * supported defaults date format or a user custom date format (set with method {@link #withDateFormat(DateFormat)}).
+   * <p/>
+   * Beware that the default formats are expressed in the current local timezone.
+   * <p/>
+   * Example:
+   *
+   * <pre><code class='java'>
+   * // assertion will pass
+   * assertThat(date).hasSameTimeAs("2002-12-17T13:01:02");
+   * assertThat(date).hasSameTimeAs("2002-12-18T13:01:02");
+   *
+   * // assertion will fail
+   * assertThat(date).hasSameTimeAs("2002-12-17T13:01:03");
+   * assertThat(date).hasSameTimeAs("2002-12-17T14:01:02");
+   * </code></pre>
+   * Defaults date format (expressed in the local time zone) are :
+   * <ul>
+   * <li><code>yyyy-MM-dd'T'HH:mm:ss.SSS</code></li>
+   * <li><code>yyyy-MM-dd'T'HH:mm:ss</code></li>
+   * <li><code>yyyy-MM-dd</code></li>
+   * </ul>
+   * <p/>
+   * Example of valid string date representations:
+   * <ul>
+   * <li><code>2003-04-26T03:01:02.999</code></li>
+   * <li><code>2003-04-26T13:01:02</code></li>
+   * <li><code>2003-04-26</code></li>
+   * </ul>
+   *
+   * @param dateAsString the given Date represented as String in default or custom date format.
+   * @return this assertion object.
+   * @throws AssertionError if the actual {@code Date} is {@code null}.
+   * @throws NullPointerException if given date as String is {@code null}.
+   * @throws AssertionError if the actual {@code Date} time is not equal to the time from date represented as
+   *           String.
+   * @throws AssertionError if the given date as String could not be converted to a Date.
+   * @author Michal Kordas
+   */
+  public S hasSameTimeAs(String dateAsString) {
+	dates.hasSameTimeAs(info, actual, parse(dateAsString));
```

Comment:
the implementation is good, I think we could improve the error message which looks like:
```
Expecting
 <2003-04-26T13:01:02>
to have time:
 <1051318861000L>
but was:
 <1051318862000L>
```
I would prefer a message that shows we are comparing the time's of two dates, for example

```
Expecting
 <2003-04-26T13:01:01>
and:
 <2003-04-26T13:01:02>
to represent the same time but was not.
Actual's time was:
 <1051318861000L>
and expected:
 <1051318862000L>
```